src/module/functions/local_function/context.rs

```diff
@@ -48,6 +49,9 @@ pub struct FunctionContext<'a> {
     /// The locals for this function's module.
     pub locals: &'a mut ModuleLocals,
 
+    /// The memories for this function's module.
+    pub memories: &'a ModuleMemories,
```

Comment:
I was a little uneasy about this as well, do you think it's possible pass down a whole `&mut Module` to house all this information?

Reply:
I think that makes sense. I am a little concerned that it might cause us to run into borrow issues where we want part of the module shared and part exclusive, but it it works then might as well roll with it.